src/address.js

```diff
@@ -7,13 +7,15 @@ var networks = require('./networks')
 var typeforce = require('typeforce')
 var types = require('./types')
 
-function fromBase58Check (address) {
-  var payload = bs58check.decode(address)
-
+function fromBase58Check (address, network) {
+  network = network || networks.bitcoin;
+  var payload = bs58check.decode(address, network.pubKeyHash)
+  
   // TODO: 4.0.0, move to "toOutputScript"
   if (payload.length < 21) throw new TypeError(address + ' is too short')
-  if (payload.length > 22) throw new TypeError(address + ' is too long')
-
+  var maxLength = network.pubKeyHash < 256 ? 21 : 22;
+  if (payload.length > maxLength) throw new TypeError(address + ' is too long')
```

Comment:
Nice,  though I'd use `0xff` to be explicit about the significance of `256`.  
You can also do `let maxLength = types.UInt8(network.pubKeyHash) ? 21 : 22`

Reply:
Updated per your suggestion.